tinygrad/tensor.py

```diff
@@ -153,12 +153,31 @@ def empty(*shape, **kwargs):
     return Tensor._loadop(LoadOps.EMPTY, prod((shape:=argfix(*shape))), **kwargs).reshape(shape)
 
   _seed: int = int(time.time())
+  _rng_counter: Optional[Tensor] = None
   @staticmethod
-  def manual_seed(seed=0): Tensor._seed = seed
+  def manual_seed(seed=0): Tensor._seed, Tensor._rng_counter = seed, Tensor([0], dtype=dtypes.uint32, requires_grad=False)
 
   @staticmethod
   def rand(*shape, **kwargs):
-    return Tensor._loadop(LoadOps.CUSTOM, prod((shape:=argfix(*shape))), arg=custom_random, **kwargs).reshape(shape)
+    if Tensor._rng_counter is None: Tensor._rng_counter = Tensor([0], dtype=dtypes.uint32, requires_grad=False)
+    num = prod((shape:=argfix(*shape)))
+    if num == 0: return Tensor.zeros(shape, **kwargs)
+    counts = (Tensor.arange(num, dtype=dtypes.uint32, requires_grad=None if kwargs.get("requires_grad") is None else kwargs.get("requires_grad")) + Tensor._rng_counter).realize()
+    if (odd_counts := num % 2): counts = counts.cat(Tensor([0], dtype=dtypes.uint32, requires_grad=False))
```

Comment:
Just padding here should work

Reply:
its actually easier just to do `if (odd_counts := num % 2): num += 1` but then it doesn't match the jax impl.